### PR TITLE
MAID-1366 BREAKS ROUTING BOOTSTRAPPING crust 3 refactor

### PR DESCRIPTION
**Pull request by inetic**
[Review on Reviewable](https://reviewable.io/reviews/maidsafe/routing/704)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @brian-js

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: Cargo.toml [hunk 1]
```diff
@@ -22,7 +22,10 @@ itertools = "*"
 log = "*"
 env_logger = "*"
 sodiumoxide = "*"
-crust = "0.2.*"
+#crust = "0.2.*"
+
+[dependencies.crust]
+path = "/home/peter/work/rust/crust"
```
Comment (benjaminbollen) [inline Cargo.toml]: yeah, that's not going to work well on the CI machines ;)

File: Cargo.toml [hunk 2]
```diff
@@ -22,7 +22,11 @@ itertools = "*"
 log = "*"
 env_logger = "*"
 sodiumoxide = "*"
-crust = "0.2.*"
+#crust = "0.2.*"
+
+[dependencies.crust]
+branch = "pub-proto"
```
Comment (benjaminbollen) [inline Cargo.toml]: at what point will this be merged into master and published to crates.io?

Comment (inetic) [inline Cargo.toml]: Changed to point to maidsafe's git repo, going to discuss with Viv adding it to crates.

File: src/routing_node.rs [hunk 3]
```diff
@@ -74,15 +73,27 @@ impl RoutingNode {
...
 
         let (crust_sender, crust_receiver) = ::std::sync::mpsc::channel::<::crust::Event>();
-        let cm = ::crust::ConnectionManager::new(crust_sender);
-        let accepting_on = cm.get_own_endpoints();
+        let mut crust_service = match ::crust::Service::new(crust_sender) {
+            Ok(service) => service,
+            Err(what) => panic!(format!("Unable to start crust::Service {}", what)),
+        };
+
+        let accepting_on = crust_service.start_default_acceptors().into_iter()
```
Comment (benjaminbollen) [inline src/routing_node.rs]: is this now a non-blocking call, but with the iterator it is made blocking, correct?

Comment (inetic) [inline src/routing_node.rs]: No, the `Service::start_default_acceptors()` returns simply a `Vec<io::Result<Endpoint>>` which are the (local) endpoints crust is now accepting on.

File: src/routing_node.rs [hunk 4]
```diff
@@ -939,6 +963,13 @@ impl RoutingNode {
         }
     }
 
+    fn connect(&mut self, endpoints: &Vec<::crust::Endpoint>) {
+        for e in endpoints {
+            self.connect_requests.insert(e.clone());
```
Comment (benjaminbollen) [inline src/routing_node.rs]: this is not okay :).  We do not want to keep any state on ::connect; but I think I can easily enough fix it, so I'll raise an issue and take it out afterwards

**Pull request merged**